### PR TITLE
Source specific interface

### LINKED ISSUE #1: Add source-specific interface
#### Description
We need interfaces that allow downloading files in GTFS format in a predictable manner.  
Each interface should work with one source. The interface should return files in a container that is independent of the source.

#### What needs to be done:
- [x] Data containers for file contents.
- [x] Interface for [London](https://www.londontransit.ca/open-data/).
- [x] Interface for [Krakow](https://gtfs.ztp.krakow.pl/) (buses).
- [x] Interface for [Krakow](https://gtfs.ztp.krakow.pl/) (trams) .

#### Extra remarks:
- Krakow's feed consists of two feeds - one for trams and one for busses. The former is suffixed with "_A" and the latter with "_T".
- [x] Files are not downloaded if their ETags are not changed.
- [x] GTFS real-time files are downloaded as Protocol Buffers, and not in the json format.

### PR DESCRIPTION
**Pull request by remilvus**
Closes #1

### CODE REVIEW

File: src/public_transport/collection/source.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            )
+        return result.returned_value
+
+    def _download_files(self) -> Tuple[data.TransitFile]:
+        files: List[Optional[data.TransitFile]] = [
+            self._downloader.get_file(category=category)
+            for category in data.TransitCategory
+        ]
+        return typing.cast(
+            Tuple[data.TransitFile], tuple(f for f in files if f is not None)
```
Comment (Gazda70) [inline src/public_transport/collection/source.py:46]: Perhaps name the variable in list comprehension like "file" so the code is more readable.

File: tests/test_source_components.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+
+def test_source_no_files(transit_source_no_files):
+    assert transit_source_no_files.request_files() == tuple()
+
+
+def test_source_request_files(transit_links, transit_source):
+    files = transit_source.request_files()
+    assert len(files) == 4
+    assert {f.category for f in files} == set(ptc.data.TransitCategory)
```
Comment (Gazda70) [inline tests/test_source_components.py:95]: The same as above, I would like to transform it to: file.category for file in files.

File: src/public_transport/collection/source.py (hunk 1 above)
Comment (remilvus) [reply src/public_transport/collection/source.py:46]: I think it's good as is, as "f" is a very common abbreviation for "file". Additionally, "file" and "files" are very similar which may be confusing.

File: tests/test_source_components.py (hunk 2 above)
Comment (remilvus) [reply tests/test_source_components.py:95]: It's a good catch in general, but I don't think it applies in this situation for the same reasons as above.

**Review by Gazda70 — COMMENTED** [review]
Approved

**Review by Gazda70 — APPROVED** [review]
Valid for merge.

**Pull request merged**